### PR TITLE
Move path implementations into `sys`

### PR DESCRIPTION
**Pull request by joboet**
Part of #117276.

r? @ChrisDenton

### CODE REVIEW

File: library/std/src/sys/path/sgx.rs
Comment (joboet) [inline library/std/src/sys/path/sgx.rs:1]: I don't see why SGX can't just use the UNIX implementation, seeing that it is also used by other platforms without filesystem support, but that's for another PR, perhaps.

File: library/std/src/sys/pal/zkvm/mod.rs [hunk 1]
```diff
@@ -24,10 +24,6 @@ pub mod net;
 #[path = "../unsupported/once.rs"]
 pub mod once;
 pub mod os;
-#[path = "../unix/os_str.rs"]
-pub mod os_str;
```
Comment (joboet) [inline library/std/src/sys/pal/zkvm/mod.rs:28]: Drive-by change, this file doesn't exist any more.

File: library/std/src/sys/pal/sgx/mod.rs
Comment (joboet) [inline library/std/src/sys/pal/sgx/mod.rs:1]: I don't see why SGX can't just use the UNIX implementation when other targets without filesystem support can, but that's a topic for another time...

Comment (ChrisDenton) [reply library/std/src/sys/pal/sgx/mod.rs:1]: Agreed. I think these PRs should be focused on moving things. Fixing everything can be done separately.
